kernel/kernel-impl/src/main/java/org/sakaiproject/util/impl/LinkMigrationHelperImpl.java

```diff
@@ -150,4 +160,63 @@ private String findLinkContent(String link) throws Exception {
 		return link.substring(contentStart+1, contentEnd);
 	}
 
+	/**
+	 * Parses an HTML content, extracts the URLs and expands the shortened ones.
+	 * This method is used mostly when importing content from other sites
+	 * @param msgBody
+	 * @return String the msgBody with the expanded URLs
+	 */
+	private String expandShortenedUrls(String msgBody){
+		String replacedBody = msgBody;
+		if(StringUtils.isNotEmpty(msgBody)){
+			Document doc = Jsoup.parse(msgBody);
+
+			Elements links = doc.select("a[href]");
+			Elements media = doc.select("[src]");
+			Elements imports = doc.select("link[href]");
+			List<String> references = new ArrayList<String>();
+			// href ...
+			for (Element link : links) {
+				references.add(link.attr("abs:href"));
+			}
+
+			// img ...
+			for (Element src : media) {
+				references.add(src.attr("abs:src"));
+			}
+
+			// js, css, ...
+			for (Element link : imports) {
+				references.add(link.attr("abs:href"));
+			}
+
+			for(String reference : references){
+				//If doesn't contain the prefix /x/, should ignore the URL.
+				if(!reference.contains("/x/")){
+					continue;
+				}
+				String longUrl = this.expandShortenedUrl(reference);
+				replacedBody = StringUtils.replace(replacedBody, reference, longUrl);
+			}
+		}
+		return replacedBody;
+	}
+
+	private String expandShortenedUrl(String shortenedUrl){
+		String expandedURL = StringUtils.EMPTY;
+		try{
+			URL url = new URL(shortenedUrl);
+			// open connection
+			HttpURLConnection httpURLConnection = (HttpURLConnection) url.openConnection(Proxy.NO_PROXY); 
```

Comment:
What if a proxy is being used?

Reply:
You're right @ottenhoff , thanks for the reminder!

Removed the NO_PROXY configuration. Do we need to implement something if a proxy is defined in the JVM properties? I don't think we need but just asking...